Add setup section (#464)
software-mansion/starknet.py: docs/setup.rst

> Account setup

Reviewer: Account creation?

Also change the name of file from setup to the title of this section in snake case.

=============

Creating an account on StarkNet
-------------------------------

> An account is needed to start interacting with StarkNet. There are a few ways of creating one:

Reviewer: `If you don't have one there are a few ways of creating one programmatically`

 - using DeployAccount transaction

>  - deploy through Cairo syscall (another account is needed)

Reviewer: I think we should mention UDC here, even if it is not ready yet.

Reviewer 2: I'd say it's a bit confusing to the users to read the guide for a library and see mentions to features that not exist and are not supported at all.

> The first approach is recommended since it doesn't rely on third-party contracts.

Reviewer: Please add an explanation how DeployAccount tx works.

> The first approach is recommended since relying on third-party contracts is not necessary.

Reviewer 2: ```suggestion
The first approach is recommended since it doesn't rely on third-party contracts.
```

Deploying an account with DeployAccount transaction requires the following:

 - class_hash of the account contract

>  - generating an account private key and deployment salt

Reviewer: a private key

>  - computing an address based on the account's secrets

Reviewer 2: Is secrets standard terminology for this? Perhaps we should just write `account private key and deployment salt` or something along the lines?

Author: Not sure, maybe your suggestion will be more readable

>  - pre-founding an address with the fee tokens (e.g. using the bridge)

Reviewer: pre**fund**ing

 - creating and signing a DeployAccount transaction with generated secrets

>  - sending the transaction to the StarkNet

Reviewer 2: ```suggestion
 - sending the transaction to StarkNet
```
I think it's not necessary?
